x/ccv/child/keeper/genesis.go

```diff
@@ -38,6 +43,31 @@ func (k Keeper) InitGenesis(ctx sdk.Context, state *types.GenesisState) {
 		// set parent client id.
 		k.SetParentClient(ctx, clientID)
 	} else {
+		// verify that latest consensus state on parent client matches the initial validator set of restarted chain
+		// thus, IBC genesis MUST run before CCV child genesis
```

Comment:
Just wondering, could this be verified by looking at the genesis.json alone? That would eliminate that condition, not that it matters.

Reply:
You can do it for the initial child chain genesis on `ValidateGenesis` which I do.

For restart child chain genesis, we need to read the state of the IBC module. There isn't a way for ValidateGenesis to look at another module's genesis. So we need to look at the already initialized IBC state in `InitGenesis` to do our verification.